1.train_models/plate_5_model_training/plate_5_interpretable_models/nbconverted/plate_5_normalized_qc_optimize_xgb.py

```diff
@@ -0,0 +1,327 @@
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define paths
+
+# ### Input paths
+
+# In[3]:
+
+
+# Load normalized qc data
+platedf_path = pathlib.Path(root_dir / "nf1_painting_repo/3.processing_features/data/single_cell_profiles/Plate_5_sc_normalized_qc.parquet").resolve(strict=True)
+
+platedf = pd.read_parquet(platedf_path)
+
+# Access spliting utilities
+sys.path.append(f"{root_dir}/1.train_models/utils")
+feat_cols = load("selected_features_1_qc_prefs.joblib")
+from GreedySubsetSum import GreedySubsetSum
+
+# ### Output paths
+
+# In[4]:
+
+
+models_path = pathlib.Path("models")
+models_path.mkdir(parents=True, exist_ok=True)
+
+data_path = pathlib.Path("data")
+data_path.mkdir(parents=True, exist_ok=True)
+
+
+# # Define Functions
+
+# In[5]:
+
+
+def filter_features(X_data):
+    """
+    Parameters
+    ----------
+    X_data: pandas Dataframe
+        Contains the feature data to be filtered.
+
+    Returns
+    -------
+    X_data: pandas Dataframe
+        The features filtered from the data.
+    """
+
+    X_data.drop(columns=X_data.filter(like="ZernikePhase").columns, inplace=True)
+    X_data.drop(columns=X_data.filter(like="ZernikeMagnitude").columns, inplace=True)
+    X_data.drop(columns=X_data.filter(like="FracAtD").columns, inplace=True)
+
+    return X_data
+
+
+# In[6]:
+
+
+def down_sample_by_genotype(df):
+    """
+    Parameters
+    ----------
+    df: pandas Dataframe
+        Cell data to be down-sampled by genotype.
+
+    Returns
+    -------
+    pandas Dataframe
+        Cell data down-sampled by genotype.
+    """
+
+    min_gene = df[gene_column].value_counts().min()
+    return (df.groupby(gene_column, group_keys=False)
+            .apply(lambda x: x.sample(n=min_gene, random_state=0))
+            )
+
+
+# # Sample Cells
+
+# ## Remove nan Cells
+
+# In[7]:
+
+
+platedf.dropna(inplace=True)
+
+
+# ## Genotype and Well groups
+
+# In[8]:
+
+
+gene_column = "Metadata_genotype"
+meta_cols = platedf.filter(like="Metadata").columns
+
+welldf = (
+    platedf.groupby(["Metadata_genotype", "Metadata_Well"])
+    .size().reset_index(name="Metadata_cell_count")
+)
+
+
+# ## Split Data
+
+# ### Seperate the test (holdout) cell data
+# The number of cells in the train-validation set is maximized by minimizing the number of cells sampled from the adjusted minority class for a desired number of holdout wells.
+# This also adjusts the number of cells in the holdout wells for the classes other than the minority class to minimize the number of cells that would otherwise not be included in any of the datasets.
+
+# In[9]:
+
+
+kwargs = {
```

Comment:
consider increasing clarity of this dictionary by improving variable name. What kind of kwargs are they? Perhaps `model_variable_kwargs` is a more appropriate name?

Reply:
oh, maybe `greedy_subset_well_kwargs`?